docs/usage.rst

```diff
@@ -149,18 +152,34 @@ Many batch systems and e.g. MPI can set a process mask on the application to res
-available in the mask is not allowed. Using ``--pika:threads=all`` will use all the hyperthreads in
-the process mask.
-
-The process mask can explicitly be ignored with the option ``--pika:ignore-process-mask`` or
-overridden with ``--pika:process-mask``. With ``--pika:ignore-process-mask`` pika behaves as if no
-process mask is set. ``--pika:process-mask`` takes an explicit hexadecimal string (beginning with
-``0x``) representing the process mask to use. The mask can also be set with the environment variable
-``PIKA_PROCESS_MASK``. ``--pika:process-mask`` takes precedence over ``PIKA_PROCESS_MASK``.
-``--pika:print-bind`` can be used to verify that the bindings used by pika are correct. Exporting
-the environment variable ``PIKA_PRINT_BIND`` (any value) is equivalent to using the
-``--pika:print-bind`` option.
+per core in the process mask. Setting the number of threads to a number higher than the number of
+cores available in the mask is not allowed. Using ``all`` as the number of threads will use all the
+hyperthreads in the process mask.
+
+The process mask can explicitly be ignored with the environment variable
+``PIKA_IGNORE_PROCESS_MASK=1`` or the command line option ``--pika:ignore-process-mask``. A process
+mask set on the process can explicitly be  overridden with the environment variable
+``PIKA_PROCESS_MASK`` or the command line option ``--pika:process-mask``. When the process mask is
+ignored, pika behaves as if no process mask is set and all cores or hyperthreads can be used by the
+runtime. ``PIKA_PROCESS_MASK`` and ``--pika:process-mask`` take an explicit hexadecimal string
+(beginning with ``0x``) representing the process mask to use. ``--pika:print-bind`` can be used to
+verify that the bindings used by pika are correct. Exporting the environment variable
+``PIKA_PRINT_BIND`` (any value) is equivalent to using the ``--pika:print-bind`` option.
+
+pika binds (or pins) worker threads to cores by default (except on macOS where thread binding is not
+supported) to avoid threads being scheduled on different cores, generally improving performance.
+Thread binding can be disabled by setting the environment variable ``PIKA_BIND`` or the command line
+option ``--pika:bind`` to the value ``none``. Threads will in this case not be bound to any
+particular core and are free to migrate between cores. This is not recommended for most use cases,
+but can be beneficial e.g. if the system is oversubscribed and threads from different processes
+would otherwise be competing for time on the same core. The default value for the binding option is
+``balanced``, which will bind threads to cores in a "balanced" way, placing threads on consecutive
+cores, avoiding the use of hyperthreads (if available). A value of ``compact`` will fill all
+hyperthreads on a core with worker threads before filling the next core.
+
+.. note::
+   The ``PIKA_THREADS``, ``PIKA_IGNORE_PROCESS_MASK``, and ``PIKA_BIND`` environment variables were
+   added in 0.32.0.
```

Comment:
Maybe it is worth adding here a second note as above? "Command line options always take precedence over environment variables." I'm thinking about the possibility of linking specifically to a given section when documenting pika usage elsewhere, so that the notes in other sections might be missed.

Reply:
I duplicated the same note as in the other section.